components/director/src/config.rs

```diff
@@ -98,23 +99,61 @@ impl Config {
     fn lookup_service_param(toml: &toml::Value,
                             service_name: &str,
                             param_name: &str)
-                            -> Option<String> {
+                            -> Result<Option<String>> {
```

Comment:
ok, `lookup_service_param()` has been updated to only accept string types, which makes the `Result<Option<String>>` the correct return type.